sharepoint-adfs/azuredeploy.json

```diff
@@ -780,27 +890,322 @@
               "UserName": "[parameters('spSvcUserName')]",
-              "Password": "[parameters('spSvcPassword')]"
+              "Password": "[variables('generalSettings').spSvcPassword]"
             },
             "SPAppPoolCreds": {
               "UserName": "[parameters('spAppPoolUserName')]",
-              "Password": "[parameters('spAppPoolPassword')]"
+              "Password": "[variables('generalSettings').spAppPoolPassword]"
             },
             "SPPassphraseCreds": {
               "UserName": "Passphrase",
-              "Password": "[parameters('spPassphrase')]"
+              "Password": "[variables('generalSettings').spPassphrase]"
+            },
+            "SPSuperUserCreds": {
+              "UserName": "[variables('generalSettings').spSuperUserName]",
+              "Password": "[parameters('serviceAccountsPassword')]"
+            },
+            "SPSuperReaderCreds": {
+              "UserName": "[variables('generalSettings').spSuperReaderName]",
+              "Password": "[parameters('serviceAccountsPassword')]"
             }
           }
         }
       }
+    },
+    {
+      "condition": "[equals(parameters('addFrontEndToFarm'), 'Yes')]",
+      "name": "[variables('vmFE').vmPublicIPName]",
+      "type": "Microsoft.Network/publicIPAddresses",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-06-15",
+      "dependsOn": [],
+      "tags": {
+        "displayName": "[variables('vmFE').vmPublicIPName]"
+      },
+      "properties": {
+        "publicIPAllocationMethod": "Dynamic",
+        "dnsSettings": {
+          "domainNameLabel": "[variables('vmFE').vmPublicIPDnsName]"
+        }
+      }
+    },
+    {
+      "condition": "[equals(parameters('addFrontEndToFarm'), 'Yes')]",
+      "name": "[variables('vmFE').vmNicName]",
+      "type": "Microsoft.Network/networkInterfaces",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-06-15",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/virtualNetworks', variables('networkSettings').vNetPrivateName)]",
+        "[resourceId('Microsoft.Network/publicIPAddresses', variables('vmFE').vmPublicIPName)]"
+      ],
+      "tags": {
+        "displayName": "[variables('vmFE').vmNicName]"
+      },
+      "properties": {
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "privateIPAllocationMethod": "Dynamic",
+              "subnet": {
+                "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('networkSettings').vNetPrivateName, variables('networkSettings').vNetPrivateSubnetSPName)]"
+              },
+              "publicIPAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIPAddresses', variables('vmFE').vmPublicIPName)]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "condition": "[equals(parameters('addFrontEndToFarm'), 'Yes')]",
```

Comment:
Use implicit?

Reply:
Can you please explain what you mean by implicit in this context?

Reply:
Same as above, the disk resource will be automatically created if needed, removes the resource from the template and also the need for the condition

Reply:
Explicit disk discussed above.
About this condition, it is not specific to this "Microsoft.Compute/disks", it's used for all 
resources required for the VM.